libdnf/conf/option_bool.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright (C) 2018-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "libdnf/conf/option_bool.hpp"
+
+#include <sstream>
+
+namespace libdnf {
+
+OptionBool::OptionBool(const OptionBool & src) : Option(src), default_value(src.default_value), value(src.value) {
+    if (src.true_values) {
+        true_values = std::make_unique<std::vector<std::string>>(*src.true_values);
+    }
+    if (src.false_values) {
+        false_values = std::make_unique<std::vector<std::string>>(*src.false_values);
+    }
+}
+
+OptionBool & OptionBool::operator=(const OptionBool & src) {
+    if (this == &src) {
+        return *this;
+    }
+    if (src.true_values) {
+        true_values = std::make_unique<std::vector<std::string>>(*src.true_values);
+    }
+    if (src.false_values) {
+        false_values = std::make_unique<std::vector<std::string>>(*src.false_values);
+    }
+    default_value = src.default_value;
+    value = src.value;
+    return *this;
+}
+
+
+OptionBool::OptionBool(
+    bool default_value, const std::vector<std::string> & true_vals, const std::vector<std::string> & false_vals)
+    : Option(Priority::DEFAULT)
+    , true_values(std::make_unique<std::vector<std::string>>(true_vals))
+    , false_values(std::make_unique<std::vector<std::string>>(false_vals))
+    , default_value(default_value)
+    , value(default_value) {}
+
+OptionBool::OptionBool(bool default_value)
+    : Option(Priority::DEFAULT)
+    , default_value(default_value)
+    , value(default_value) {}
+
+bool OptionBool::from_string(const std::string & value) const {
+    auto tmp_value = value;
+    for (auto & ch : tmp_value) {
+        ch = static_cast<char>(std::tolower(static_cast<unsigned char>(ch)));
```

Comment:
Please could add a comment about the transformation?

Reply:
Yes, of course. I will add.